iothub_client/tests/iothubtransport_amqp_common_ut/iothubtransport_amqp_common_ut.c

```diff
@@ -4329,6 +4348,137 @@ TEST_FUNCTION(ConnectionStatusCallBack_UNAUTH_msg_communication_error)
     destroy_transport(handle, device_handle, NULL);
 }
 
+TEST_FUNCTION(Transport_Fully_Reconnects_After_5_AMQP_device_errors)
+{
+    // arrange
+    initialize_test_variables();
+    time_t current_time = TEST_current_time;
+    bool is_state_change_timedout = false;
+    TRANSPORT_LL_HANDLE handle = create_transport();
+    IOTHUB_DEVICE_CONFIG* device_config = create_device_config(TEST_DEVICE_ID_CHAR_PTR, true);
+
+    IOTHUB_DEVICE_HANDLE device_handle;
+    device_handle = register_device(handle, device_config, &TEST_waitingToSend, true);
+
+    crank_transport_ready_after_create(handle, &TEST_waitingToSend, 0, false, true, 1, TEST_current_time, false);
+
+    // act
+    // Fail AMQP device enough times to trigger reconnection:
+    umock_c_reset_all_calls();
+
+    for (int i = 0; i < 4; i++)
```

Comment:
MAX_NUMBER_OF_DEVICE_FAILURES but that is defined in another .c file

Reply:
Yeah, I should define a macro to reference it, at least. I'll do it.